src/Spatial/Euclidean/LineSegment2D.cs

```diff
@@ -159,8 +159,17 @@ public bool TryIntersect(LineSegment2D other, out Point2D intersection, Angle to
             var t = (q - p).CrossProduct(s) / r.CrossProduct(s);
             var u = (p - q).CrossProduct(r) / s.CrossProduct(r);
 
-            intersection = p + (t * r);
-            return (0.0 <= t && t <= 1.0) && (0.0 <= u && u <= 1.0);
+            var isIntersected = (0.0 <= t && t <= 1.0) && (0.0 <= u && u <= 1.0);
+            if (isIntersected)
+            {
+                intersection = p + (t * r);
+                return true;
+            }
+            else
```

Comment:
You can simplify all this by simply ```return intersection != null;``` and completely removing the ```else``` case.

Reply:
Thanks for a comment.

How about the following case?
![83840266-f4d1c880-a6b2-11ea-8137-56f9fee366af](https://user-images.githubusercontent.com/1047493/228208933-d9c76e42-89d1-4c0b-9ee5-76a89d599c38.png)

This comes from #178.

Reply:
Thanks for a comment.
Your understanding is correct that we can verify it by using `LineSegment2DTests.IntersectWithTest2`.

I'm happy with the merged code. Thanks.

Reply:
Thank YOU for opening the issue and providing a PR 😃